grype/db/v6/vulnerability_store.go

```diff
@@ -118,7 +119,40 @@ func (s *vulnerabilityStore) AddVulnerabilities(vulnerabilities ...*Vulnerabilit
 			}
 		}
 
-		// write the vulnerability handle to the DB
+		if err := s.addUniqueVulnerability(v); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *vulnerabilityStore) addUniqueVulnerability(v *VulnerabilityHandle) error {
+	// if this is a unique (name, status, published, modified, withdrawn, provider_id, blob_id) then it will be created
+	// otherwise do not create the new entry (this is to prevent duplicates)
+	query := s.db.Where("name = ? AND status = ? AND provider_id = ? AND blob_id = ?", v.Name, v.Status, v.ProviderID, v.BlobID)
+
+	if v.PublishedDate != nil {
+		query = query.Where("published_date = ?", *v.PublishedDate)
+	} else {
+		query = query.Where("published_date IS NULL")
+	}
+
+	if v.ModifiedDate != nil {
+		query = query.Where("modified_date = ?", *v.ModifiedDate)
+	} else {
+		query = query.Where("modified_date IS NULL")
+	}
+
+	if v.WithdrawnDate != nil {
+		query = query.Where("withdrawn_date = ?", *v.WithdrawnDate)
+	} else {
+		query = query.Where("withdrawn_date IS NULL")
+	}
```

Comment:
Should this only be checking a subset of these fields to determine if the record already exists, like name and provider ID? I could certainly see having multiple records for the same ID, but if the only difference is modified_date, would this be a different record?